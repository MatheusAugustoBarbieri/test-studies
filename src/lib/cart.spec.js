import Cart from './cart.js';
describe('cart', () => {
  const cart = new Cart();
  const product = {
    title: 'Tenis',
    price: 35388,
  };
  const product2 = {
    title: 'Tenis',
    price: 41872,
  };
  describe('getTotal', () => {
    it('verifica se o getTotal() existe e foi executado', () => {
      expect(cart.getTotal()).toEqual(0);
    });
    it('deve multiplicar quantidade e preÃ§o e retornar o total', () => {
      const item = {
        product,
        quantity: 2,
      };
      cart.add(item);
      expect(cart.getTotal(item)).toEqual(70776);
    });
    it('nao mais do que um produto exista', () => {
      cart.add({
        product,
        quantity: 1,
      });
      cart.add({
        product,
        quantity: 2,
      });
      cart.add({
        product,
        quantity: 2,
      });
      expect(cart.getTotal()).toEqual(70776);
    });
    it('garantir q continua funcionando se voce remover um produto', () => {
      cart.add({
        product,
        quantity: 2,
      });
      cart.add({
        product: product2,
        quantity: 1,
      });
      cart.remove(product);
      expect(cart.getTotal()).toEqual(41872);
    });
  });
  describe('checkout', () => {
    it('deve retornar um objeto com o total e a lista de itens', () => {
      cart.add({
        product,
        quantity: 3,
      });
      cart.add({
        product: product2,
        quantity: 1,
      });
      expect(cart.checkout()).toMatchSnapshot();
    });
    it('deve limpar o carrinho quando o usuario ir para o checkout', () => {
      cart.add({
        product: product2,
        quantity: 3,
      });
      cart.checkout();
      expect(cart.getTotal()).toEqual(0);
    });
    it('summary deve retornar valores que sejam maiores que 0', () => {
      cart.add({
        product,
        quantity: 3,
      });
      cart.add({
        product: product2,
        quantity: 5,
      });

      expect(cart.summary()).toMatchSnapshot();
      expect(cart.getTotal()).toBeGreaterThan(0);
    });
  });
});
