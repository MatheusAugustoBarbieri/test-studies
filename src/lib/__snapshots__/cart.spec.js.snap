// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart checkout deve retornar um objeto com o total e a lista de itens 1`] = `
Object {
  "itens": Array [
    Object {
      "product": Object {
        "price": 35388,
        "title": "Tenis",
      },
      "quantity": 3,
    },
    Object {
      "product": Object {
        "price": 41872,
        "title": "Tenis",
      },
      "quantity": 1,
    },
  ],
  "total": 148036,
}
`;

exports[`cart checkout summary deve retornar valores que sejam maiores que 0 1`] = `
Object {
  "itens": Array [
    Object {
      "product": Object {
        "price": 35388,
        "title": "Tenis",
      },
      "quantity": 3,
    },
    Object {
      "product": Object {
        "price": 41872,
        "title": "Tenis",
      },
      "quantity": 5,
    },
  ],
  "total": 315524,
}
`;

exports[`cart checkout summary deve retornar valores que sejam maiores que 0 1`] = `
Object {
  "itens": Array [
    Object {
      "product": Object {
        "price": 35388,
        "title": "Tenis",
      },
      "quantity": 3,
    },
    Object {
      "product": Object {
        "price": 41872,
        "title": "Tenis",
      },
      "quantity": 5,
    },
  ],
  "total": 315524,
}
`;
